packages/scene-composer/src/components/toolbars/floatingToolbar/items/AddObjectMenu.tsx

```diff
@@ -276,7 +277,24 @@ export const AddObjectMenu = (): JSX.Element => {
           uri: modelUri,
           modelType: modelType ?? ext.toUpperCase(),
         };
-
+        if (ext.toUpperCase() == 'GLTF') {
+          const animationComponent: IAnimationComponent = {
+            currentAnimations: [],
+            type: 'Animation',
+            uri: modelUri,
+          };
+          const node = {
+            name: filename,
+            components: [gltfComponent, animationComponent],
+            parentRef: mustBeRoot ? undefined : getRefForParenting(),
+          } as unknown as ISceneNodeInternal;
+          if (enhancedEditingEnabled && !modelType) {
+            setAddingWidget({ type: KnownComponentType.ModelRef, node });
+          } else {
+            appendSceneNode(node);
+          }
+          return;
```

Comment:
these are redundant code, you should just create animation component when it's GLTF, and when animation component is created, add it to node.components. something like 
```
let animationComponent: IAnimationComponent | undefined = undefined;
if (ext.toUpperCase() == 'GLTF') {
  animationComponent = ....
}
const node = {
  components: animationComponent ? [gltfComponent, animationComponent] : [gltfComponent]
  ...
}
```

Reply:
and should we only add animation component when the feature flag is on?